pkg/controller/worker/helper.go

```diff
@@ -59,3 +63,21 @@ func (w *workerDelegate) updateWorkerProviderStatus(ctx context.Context, workerS
 		return nil
 	})
 }
+
+func makeCopyOfMap(mp map[string]string) map[string]string {
+	var mpDeepCopy = make(map[string]string)
+	for k, v := range mp {
+		mpDeepCopy[k] = v
+	}
+	return mpDeepCopy
+}
+
+func addAwsCsiDriverTopologyLabel(poolLabels map[string]string, zone string) map[string]string {
```

Comment:
Please, add the csi labels only if the CSI is enabled for the cluster. 
You can see an example here https://github.com/gardener/gardener-extension-provider-aws/blob/0f39e3d1c1eed0075e3e5661793da74fa94feea1/pkg/webhook/controlplane/ensurer.go#L396 how to check whether it is enabled or not.